src/queries.js

```diff
@@ -64,30 +64,47 @@ export function appQuery(appbucket, limit) {
   return _.map(lines, (l) => l + ";");
 }
 
-export function browserSummaryQuery(browserbucket, windowbucket, afkbucket, limit, filterAFK) {
-  browserbucket = browserbucket.replace('"', '\\"');
+const chrome_appnames = ["Google-chrome", "chrome.exe", "Chromium", "Google Chrome", "Chromium-browser", "Chromium-browser-chromium", "Google-chrome-beta", "Google-chrome-unstable"];
+const firefox_appnames = ["Firefox", "Firefox.exe", "firefox", "firefox.exe", "Firefox Developer Edition", "Firefox Beta", "Nightly"]
+
+export function browserSummaryQuery(browserbuckets, windowbucket, afkbucket, limit, filterAFK) {
+  // Escape `"`
+  browserbuckets = _.map(browserbuckets, (b) => b.replace('"', '\\"'));
   windowbucket = windowbucket.replace('"', '\\"');
   afkbucket = afkbucket.replace('"', '\\"');
   limit = limit || 5;
-  var browser_appnames = "";
-  if (browserbucket.endsWith("-chrome")){
-    browser_appnames = JSON.stringify(["Google-chrome", "chrome.exe", "Chromium", "Google Chrome", "Chromium-browser", "Chromium-browser-chromium", "Google-chrome-beta", "Google-chrome-unstable"]);
-  } else if (browserbucket.endsWith("-firefox")){
-    browser_appnames = JSON.stringify(["Firefox", "Firefox.exe", "firefox", "firefox.exe", "Firefox Developer Edition", "Firefox Beta", "Nightly"]);
-  }
 
-  return [
-    'events = flood(query_bucket("' + browserbucket + '"));',
-    'window_browser = flood(query_bucket("' + windowbucket + '"));',
-    'window_browser = filter_keyvals(window_browser, "app", ' + browser_appnames + ');',
-  ].concat(filterAFK ? [
-    'not_afk = flood(query_bucket("' + afkbucket + '"));',
-    'not_afk = filter_keyvals(not_afk, "status", ["not-afk"]);',
-    'window_browser = filter_period_intersect(window_browser, not_afk);',
-  ] : [])
-  .concat([
-    'events = filter_period_intersect(events, window_browser);',
-    'events = split_url_events(events);',
+  // If multiple browser buckets were found
+  let code = (
+    `events = [];
+     window = flood(query_bucket("${windowbucket}"));`
+  ) + (
+    `not_afk = flood(query_bucket("${afkbucket}"));
```

Comment:
Could check for filter_afk here as well to avoid an unnecessary bucket query in that case.

Reply:
Done!